chaindexing-repos/src/index.ts

```diff
@@ -12,11 +12,19 @@ export abstract class Repo<Pool, Conn> extends Migratable<Conn> {
     conn: Conn,
     contractAddresses: UnsavedContractAddress[]
   ): Promise<void>;
-  abstract streamContractAddresses(
+  abstract getContractAddressesStream(
     conn: Conn,
-    streamer: (contractAddresses: ContractAddress[]) => Promise<void>
-  ): Promise<void>;
+    opts?: { limit?: number }
+  ): {
+    next: () => Promise<ContractAddress[] | null>;
+  };
   abstract createEvents(conn: Conn, events: Event[]): Promise<void>;
+  abstract getEventsStream(
+    conn: Conn,
+    opts?: { limit?: number }
+  ): {
+    next: () => Promise<Event[] | null>;
+  };
```

Comment:
Let's keep the interface simple. Do you have any use case for a `null` return?

```suggestion
    opts?: { limit?: number }
  ): {
    next: () => Promise<ContractAddress[]>;
  };
  abstract createEvents(conn: Conn, events: Event[]): Promise<void>;
  abstract getEventsStream(
    conn: Conn,
    opts?: { limit?: number }
  ): {
    next: () => Promise<Event[]>;
  };
```

Reply:
I just wasn't sure, if the query will return an empty array or null when the offset is beyond the range. The typings is not helping for this one.